File: mac/Classes/Model/Results/LRProjectTarget.m

#import "LRProjectTarget.h"
#import "Action.h"
#import "Project.h"
#import "LRBuild.h"


@interface LRProjectTarget ()

@end


@implementation LRProjectTarget

- (instancetype)initWithAction:(Action *)action modifiedFiles:(NSArray *)modifiedFiles {
    self = [super initWithAction:action];
    if (self) {
        _modifiedFiles = [modifiedFiles copy];
    }
    return self;
}

- (void)invokeWithCompletionBlock:(dispatch_block_t)completionBlock build:(LRBuild *)build {
    LROperationResult *result = [self newResult];
    [self.action invokeForProject:self.project withModifiedFiles:_modifiedFiles result:result completionHandler:^{
        [build addOperationResult:result forTarget:self key:[NSString stringWithFormat:@"%@.postproc", self.project.path]];
        completionBlock();
    }];
}

@end

File: mac/Classes/Model/Results/LRProjectTarget.swift
import Foundation
import LRActionKit

@objc class LRProjectTarget : LRTarget {

    let modifiedFiles: [ProjectFile]

    init(rule: Rule, modifiedFiles: [ProjectFile]) {
        self.modifiedFiles = modifiedFiles
        super.init(rule: rule)
    }

    override func invoke(#build: LRBuild, completionBlock: dispatch_block_t) {
        let result = self.newResult()
        rule.invokeWithModifiedFiles(modifiedFiles, result: result) {
            build.addOperationResult(result, forTarget: self, key:"\(self.project.path).postproc")
            completionBlock()
        }
    }

}
